### PR TITLE
Feature/1146/openapi generate

### PR DESCRIPTION
**Pull request by garyluu**
#1146 

Auto-generate openapi.yaml

Requires everyone to `npm i -g swagger2openapi`

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -32,6 +32,7 @@ env:
 
 before_install:
 - pip install --upgrade pip
+- npm install -g swagger2openapi
```
Comment (denis-yuen) [inline .travis.yml]: Document this new dependency for basic builds in the readme

File: dockstore-webservice/pom.xml [hunk 2]
```diff
@@ -945,6 +945,29 @@
...
+                <artifactId>exec-maven-plugin</artifactId>
+                <version>1.6.0</version>
+                <executions>
+                    <execution>
+                        <id>swagger2openapi)</id>
+                        <goals>
+                            <goal>exec</goal>
+                        </goals>
+                        <!--So it executes after the swagger.yaml's compile-->
+                        <phase>test</phase>
```
Comment (denis-yuen) [inline dockstore-webservice/pom.xml]: Not really a test, should probably be in a different phase. 
Note that Maven runs plugins in the same phase in the order that they appear in the pom.xml

Comment (garyluu) [reply dockstore-webservice/pom.xml]: interesting

File: README.md [hunk 3]
```diff
@@ -116,7 +116,8 @@ The following section is useful for Dockstore developers (e.g. those that want t
 The dependency environment for Dockstore is described by our 
 [Travis-CI config](https://github.com/ga4gh/dockstore/blob/develop/.travis.yml). In addition to the dependencies for 
 Dockstore users, note the setup instructions for postgres. Specifically, you will need to have postgres installed 
-and setup with the database user specified in [.travis.yml](https://github.com/ga4gh/dockstore/blob/develop/.travis.yml#L26). 
+and setup with the database user specified in [.travis.yml](https://github.com/ga4gh/dockstore/blob/develop/.travis.yml#L26).
```
Comment (denis-yuen) [inline README.md]: Worth putting in a note that we aim to have unit tests that run without postgres. 
Integration tests would require it

**Pull request merged**